README.md

```diff
@@ -241,8 +249,95 @@ $ run_model '{
   "file_list": ["/data-share/predictions/FastCharge_2_CH29_full_model_predictions.json"],
 }
 ```
+
+## Data requirements
+
+BEEP automatically parses and structures data based on specific outputs from various 
+battery cyclers. The following column headers are required for downstream processing of each 
+cycler.
+
+### Arbin
+
+##### Metadata
+
+| Field name | Explanation | Data Types |
+|------------|-------------|------------|
+
+
+
+##### Cycler Data
+
+| Column name | Required |   Explanation  | Unit |  Data Type |
+|-------------|----------|-------------|------|------------|
+| `data_point` |   |   |   |   `int32` |
+| `test_time` |   |   |   |   `float32` |
+| `datetime` |   |   |   |   `float32` |
+| `step_time` |   |   |   |   `float32` |
+| `step_index` |   |   |   |   `int16` |
+| `cycle_index` |   |   |   |   `int32` |
+| `current` |   |   |   |   `float32` |
+| `voltage` |   |   |   |   `float32` |
+| `charge_capacity` |   |   |   |   `float64` |
+| `discharge_capacity` |   |   |   |   `float64` |
+| `charge_energy` |   |   |   |   `float64` |
+| `discharge_energy` |   |   |   |   `float64` |
+| `dv/dt` |   |   |   |   `float32` |
+| `internal_resistance` |   |   |   |   `float32` |
+| `temperature` |   |   |   |   `float32` |
```

Comment:
I believe that temperature and internal resistance are optional.

Reply:
Hmm it seems Temperature is actually a required field for processedcycler run? Maybe not itentionally so. removing that column from the `2017-12-04_4_65C-69per_6C_CH29.csv` test file throws an error when ProcessedCyclerRun is instantiated via `pcr = ProcessedCyclerRun.from_raw_cycler_run(...)`

Reply:
A minimal script to reproduce:

```
from beep.structure import ProcessedCyclerRun, RawCyclerRun
import shutil
import pandas as pd


ARBIN_FILE = "/path/to/beep/beep/tests/test_files/2017-12-04_4_65C-69per_6C_CH29.csv"
ARBIN_METADATA = "/path/to/beep/beep/tests/test_files/2017-12-04_4_65C-69per_6C_CH29_Metadata.csv"


# create new input files without the temperature column
df = pd.read_csv(ARBIN_FILE)
metadf = pd.read_csv(ARBIN_METADATA)
df_removed = df.drop(columns=["Temperature"])
new_data_fname = ARBIN_FILE.replace(".csv", f"_no_Temperature.csv")
new_metadata_fname = ARBIN_METADATA.replace("_Metadata.csv", f"_no_Temperature_Metadata.csv")
df_removed.to_csv(new_data_fname, index=False)
shutil.copy(ARBIN_METADATA, new_metadata_fname)

# try to structure from new input files 
rcr = RawCyclerRun.from_arbin_file(new_data_fname)
pcr = ProcessedCyclerRun.from_raw_cycler_run(rcr)
```


Throws:

```
Traceback (most recent call last):
  File "/Users/ardunn/alex/tri/code/beep/alex_scripts/tmp.py", line 33, in <module>
    pcr = ProcessedCyclerRun.from_raw_cycler_run(rcr)
  File "/Users/ardunn/alex/tri/code/beep/beep/structure.py", line 1392, in from_raw_cycler_run
    diagnostic_available=diagnostic_available,
  File "/Users/ardunn/alex/tri/code/beep/beep/structure.py", line 289, in get_interpolated_cycles
    axis="voltage",
  File "/Users/ardunn/alex/tri/code/beep/beep/structure.py", line 234, in get_interpolated_steps
    resolution=resolution,
  File "/Users/ardunn/alex/tri/code/beep/beep/structure.py", line 1809, in get_interpolated_data
    df = dataframe.loc[:, columns]
  File "/Users/ardunn/alex/tri/code/beepenv/lib/python3.7/site-packages/pandas/core/indexing.py", line 1761, in __getitem__
    return self._getitem_tuple(key)
  File "/Users/ardunn/alex/tri/code/beepenv/lib/python3.7/site-packages/pandas/core/indexing.py", line 1288, in _getitem_tuple
    retval = getattr(retval, self.name)._getitem_axis(key, axis=i)
  File "/Users/ardunn/alex/tri/code/beepenv/lib/python3.7/site-packages/pandas/core/indexing.py", line 1953, in _getitem_axis
    return self._getitem_iterable(key, axis=axis)
  File "/Users/ardunn/alex/tri/code/beepenv/lib/python3.7/site-packages/pandas/core/indexing.py", line 1594, in _getitem_iterable
    keyarr, indexer = self._get_listlike_indexer(key, axis, raise_missing=False)
  File "/Users/ardunn/alex/tri/code/beepenv/lib/python3.7/site-packages/pandas/core/indexing.py", line 1552, in _get_listlike_indexer
    keyarr, indexer, o._get_axis_number(axis), raise_missing=raise_missing
  File "/Users/ardunn/alex/tri/code/beepenv/lib/python3.7/site-packages/pandas/core/indexing.py", line 1654, in _validate_read_indexer
    "Passing list-likes to .loc or [] with any missing labels "
KeyError: 'Passing list-likes to .loc or [] with any missing labels is no longer supported, see https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike'
```

If unable to reproduce, the problem is likely on my end

Reply:
So it looks like we have two options:


#### Option 1: enforcing imputation across all cyclers in `RawCyclerRun.__init__`

We could remove that imputation from maccor specifically, and then enforce it with a kwarg to __init__ for all `RawCyclerRuns`. This might also solve the problem with Arbin (and yet unknown issues with the other cyclers). So like the `__init__` of `RawCyclerRun` could include an `inpute_missing` argument:

```
def __init__(self, data, metadata, eis=None, validate=False, filename=None, impute_missing=True):
```

Where, if True, `RawCyclerRun` will automatically nan the columns which are required for downstream processing but don't need actual values (e.g., Temperature). 

#### Option 2: Remove the need for temperature columns altogether

If temperature is not required for the processing steps themselves, we could just remove whatever common code needs it. Then we would not need to impute any of these "nominally required" columns which are not actually required for interpolation but cause errors when missing. From looking at that stacktrace it doesn't seem there is a direct way to fix it, but maybe someone who is more familiar with that code has thoughts?

Reply:
I like Option 1 the best. I think that for most of the experiments, there should be temperature data and if there is not imputing with NaN makes the most sense.

Reply:
Ok, it turns out option 1 alone was not very robust so I wound up doing both.

Temperature was only nominally required (seemingly) because of a bad `.loc` which fails if it is missing, and separately because of `structure_dtypes.yaml` which implicitly enforced that field to be there. Ditto for internal_resistance. These issues are fixed with latest commit (though not sure if they will pass all tests yet)

Reply:
Ok, I went with option 1. It is not a great fix, but will do in the interim.